robotools/tests.py

```diff
@@ -465,6 +465,581 @@ def test_parameter_validation(self) -> None:
         evotools._prepare_aspirate_dispense_parameters(
             rack_label="WaterTrough", position=1, volume=15, forced_rack_type="valid forced rack type"
         )
+
+        # test _prepare_evo_aspirate_dispense_parameters
+        # define a labware correctly for testing purposes
+        plate = liquidhandling.Labware("DWP", 8, 12, min_volume=0, max_volume=2000, initial_volumes=1000)
+        # test labware argument checks
+        with pytest.raises(ValueError, match="Invalid labware:"):
+            evotools._prepare_evo_aspirate_dispense_parameters(
+                labware="wrong_labware_type",
+                wells=["A01", "B01"],
+                labware_position=(38, 2),
+                volume=15,
+                liquid_class="Water_DispZmax-1_AspZmax-1",
+                tips=[1, 2],
+            )
```

Comment:
Just a tip for the future: These tests can be written in a much shorter way by parametrizing with override-kwargs and the expected exception (type and) message
```python
@pytest.mark.parametrize("overrides,match", [
    ({}, None),  # no overrides, no error expected
    (dict(labware="wrong"), "Invalid labware"),
    (..., ...),
])
def test_lots_of_checks(overrides, match):
    kwargs = dict(wells=["A01", "B01"], volume=15, ...)
    kwargs.update(overrides)
    if not match:
        f(**kwargs)
    else:
        with pytest.raises(ValueError, match=match):
            f(**kwargs)
    pass
```

Reply:
Okay, thanks for the tip. That would have been quite handy, indeed.